compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -157,6 +158,11 @@ fn object_safety_violations_for_trait(
                 .in_definition_order()
                 .filter(|item| item.kind == ty::AssocKind::Type)
                 .filter(|item| !tcx.generics_of(item.def_id).params.is_empty())
+                .filter(|item| {
+                    tcx.def_path(item.def_id).data.last().map_or(true, |def_path_data| {
+                        def_path_data.data != DefPathData::ImplTraitAssocTy
+                    })
```

Comment:
We don't need to check def path data, we can check opt_rpitit_info, right? Or just use that helper method you added?

Reply:
Right, fixed.